caas/kubernetes/provider/credentials.go

```diff
@@ -149,3 +74,30 @@ func (p environProviderCredentials) RegisterCredentials(cld cloud.Cloud) (map[st
 		},
 	}, nil
 }
+
+func registerCredentialsKubeConfig(
```

Comment:
This doesn't look like it registers anything - it builds a juju cloud credential map from the cloud name and local kube config.
Maybe rename the method and add a comment

Reply:
no it makes sure that credentials we need are registered into the Kubernetes cluster. It's just a local kube conf version of the minikube one.